### ISSUE TITLE
Hook creating .secrets.baseline using detect-secrets 0.13.1 which is incompatible with 0.14.x

### ISSUE THREAD

**Issue by sengi** [alphagov/gds-pre-commit#80]
I suspect this is at least partly down to something whacky going on with Python installations on my machine, but if I do this:

1. cd into the root dir of a repo/branch which doesn't yet have a .secrets.baseline
2. `touch foo && git add foo && git commit` (or anything that invokes global precommits)

```
Unable to open baseline file: `REPO_ROOT`/.secrets.baseline
Please create it via
   detect-secrets scan > .secrets.baseline
```

3. `detect-secrets scan > .secrets.baseline`
4. `git commit`

```
...
The baseline file was updated.
Probably to keep line numbers of secrets up-to-date.
Please `git add .secrets.baseline`, thank you.
```

5. `grep version .secrets.baseline`

```
  "version": "0.13.1",
```

6. `detect-secrets --version`

```
0.14.3
```

7. `detect-secrets audit .secrets.baseline`

```
Traceback (most recent call last):
  File "/usr/local/bin/detect-secrets", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.8/site-packages/detect_secrets/main.py", line 73, in main
    audit.audit_baseline(args.filename[0])
  File "/usr/local/lib/python3.8/site-packages/detect_secrets/core/audit.py", line 95, in audit_baseline
    custom_plugin_paths=original_baseline['custom_plugin_paths'],
KeyError: 'custom_plugin_paths'
```

So it seems that the precommit hook is somehow loading an outdated version of detect-secrets from somewhere, but the version that runs from $PATH is up-to-date.

I found an old copy of detect-secrets under `/usr/local/lib/python2.7/site-packages` but I deleted it and still get the same results as above. (I made sure it's still not there as well - the old version isn't being reinstalled, at least not in that directory.)

The up-to-date detect-secrets on my machine is in `/usr/local/lib/python3.8/site-packages`.

The Python installations that I was able to find on my machine are:

* The standard (and ancient) macOS Python 2.7, installed as `/usr/bin/python` and `/usr/bin/python2.7` and having `/Library/Python/2.7/site-packages` in its sys.path (but no references to `/usr/lib` or `/usr/local/lib`).
* Python 3.8 installed via Homebrew. This is the Python which appears first in my $PATH and it's the one which has detect-secrets installed in its site-packages dir.

There's also a `/usr/local/lib/python2.7/site-packages` but I can't find a Python installation which corresponds to that - I'm guessing it's left over from an old Python 2.7 installed via Homebrew which has been deleted some time ago. That site-packages dir no longer contains a detect-secrets package but the problem persists.

🤔

**Comment by sengi** [comment]
`python -m pip list` (the system python2 AFAICT) lists a bunch of packages but there's no sign of a `detect-secrets`.

**Comment by sengi** [comment]
@0atman from our debugging session now:

```
$ find / -iname 'detect*secrets' 2>/dev/null
/usr/local/bin/detect-secrets
/usr/local/lib/python3.8/site-packages/detect_secrets
~/.cache/pre-commit/repodkp0ypy7/detect_secrets
~/.cache/pre-commit/repodkp0ypy7/py_env-python3.8/bin/detect-secrets
~/.cache/pre-commit/repodkp0ypy7/py_env-python3.8/lib/python3.8/site-packages/detect_secrets
```

So we discovered that the pre-commit framework installs the precommit hook's dependencies into a Python virtualenv and its detect-secrets version is pinned to version 0.13.1 in [gds-pre-commit/global_install/hooks/.pre-commit-config.yaml](https://github.com/alphagov/gds-pre-commit/blob/0db4a1b63a97b4535ea9b7811a8a3a8641c599f2/global_install/hooks/.pre-commit-config.yaml#L9). Meanwhile for new users, pip will install the current version of detect-secrets by default, which happens not to be backward-compatible with the pinned version in the pre-commit hook's virtualenv.

**Comment by 0atman** [comment]
@sengi omg thank you for that epic bug hunting session.

What a treat! Two learning outcomes:

1. We should make sure the hook has the same shebang as the user's installed `pre-commit` executable
2. the pre-commit framework pins detect-secrets at a certain version. We must tell the users to install only this version as a dependency.

**Comment by 0atman** [comment]
And another important thing that it took me far too long to remember: pre-commit makes its own virtualenvironment

**Comment by 0atman** [comment]
Maybe `sh` is cross-platform enough?

```shell
sh -c "head -n1 `which pre-commit`"
```
Won't work on Windows, but what does?

**Comment by 0atman** [comment]
pivoting away from our installer, this issue is very difficult to fix, so we're sidestepping it in #83
